_ml-commons-plugin/gpu-acceleration.md

```diff
@@ -0,0 +1,285 @@
+deb https://apt.repos.neuron.amazonaws.com ${VERSION_CODENAME} main
+EOF
+wget -qO - https://apt.repos.neuron.amazonaws.com/GPG-PUB-KEY-AMAZON-AWS-NEURON.PUB | sudo apt-key add -
+
+# Update OS packages
+sudo apt-get update -y
+
+################################################################################################################
+# To install or update to Neuron versions 1.19.1 and newer from previous releases:
+# - DO NOT skip 'aws-neuron-dkms' install or upgrade step, you MUST install or upgrade to latest Neuron driver
+################################################################################################################
+
+# Install OS headers
+sudo apt-get install linux-headers-$(uname -r) -y
+
+# Install Neuron Driver
+sudo apt-get install aws-neuronx-dkms -y
+
+####################################################################################
+# Warning: If Linux kernel is updated as a result of OS package update
+#          Neuron driver (aws-neuron-dkms) should be re-installed after reboot
+####################################################################################
+
+# Install Neuron Tools
+sudo apt-get install aws-neuronx-tools -y
+
+######################################################
+#   Only for Ubuntu 20 - Install Python3.7
+sudo add-apt-repository ppa:deadsnakes/ppa
+sudo apt-get install python3.7
+######################################################
+# Install Python venv and activate Python virtual environment to install    
+# Neuron pip packages.
+cd ~
+sudo apt-get install -y python3.7-venv g++
+python3.7 -m venv pytorch_venv
+source pytorch_venv/bin/activate
+pip install -U pip
+
+# Set Pip repository  to point to the Neuron repository
+pip config set global.extra-index-url https://pip.repos.neuron.amazonaws.com
+
+#Install Neuron PyTorch
+pip install torch-neuron torchvision
+# If need to trace neuron model, install torch neuron with this command
+# pip install torch-neuron neuron-cc[tensorflow] "protobuf==3.20.1" torchvision
+
+# If need to trace neuron model, install transformers for tracing Huggingface model.
+# pip install transformers
+
+# Copy torch neuron lib to OpenSearch
+PYTORCH_NEURON_LIB_PATH=~/pytorch_venv/lib/python3.7/site-packages/torch_neuron/lib/
+mkdir -p $OS_HOME/lib/torch_neuron; cp -r $PYTORCH_NEURON_LIB_PATH/ $OS_HOME/lib/torch_neuron
+export PYTORCH_EXTRA_LIBRARY_PATH=$OS_HOME/lib/torch_neuron/lib/libtorchneuron.so
+echo "export PYTORCH_EXTRA_LIBRARY_PATH=$OS_HOME/lib/torch_neuron/lib/libtorchneuron.so" | tee -a ~/.bash_profile
+
+# Increase JVm stack size to >=2MB
+echo "-Xss2m" | tee -a $OS_HOME/config/jvm.options
+# Increase max file descriptors to 65535
+echo "$(whoami) - nofile 65535" | sudo tee -a /etc/security/limits.conf
+# max virtual memory areas vm.max_map_count to 262144
+sudo sysctl -w vm.max_map_count=262144
+```
+
+#### Amazon Linux2
+
+```
+# Configure Linux for Neuron repository updates
```

Comment:
Break all of these out into individual steps, don't put them in a massive code block.

Reply:
These scripts are based on the [Neuron documentation](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/frameworks/torch/torch-neuronx/setup/pytorch-install.html), with customization's to memory to allow Neuron to be used in OpenSearch.  I'm open to breaking out the steps specific to OpenSearch, but would need to submit a PR on the Neuron docs in order to fix spelling and grammar issues within the script itself.

Reply:
A script would start with a #!/bin/bash - these are just collections of one-line commands.

Reply:
Fixed the language in the comments, but after consulting @ylwu-amzn, I've decided to keep the scripts together. We expect that system architects will copy the contents into one script file insides their cluster. Keeping them together makes completing that step easier.

Reply:
Also, our build script doesn't with `#!/bin/bash`, see https://github.com/opensearch-project/documentation-website/blob/main/build.sh